salt/modules/puppet.py

```diff
@@ -158,7 +163,7 @@ def run(*args, **kwargs):
 
     puppet.kwargs.update(salt.utils.clean_kwargs(**kwargs))
 
-    ret = __salt__['cmd.run_all'](repr(puppet), python_shell=False)
+    ret = __salt__['cmd.run_all'](repr(puppet), python_shell=puppet.useshell)
```

Comment:
Given this change, is there a chance that a person can provide malicious arguments, ala: `puppet_legit_args; do_evil_thing.sh`? I'd like to have us test this first and if it's a concern, address it.